persistence/src/commonTest/kotlin/com/wire/kalium/persistence/backup/DatabaseImporterTest.kt

```diff
@@ -631,4 +651,34 @@ class RestoreBackupTest : BaseDatabaseTest() {
 
         return conversationAdded
     }
+
+    private suspend fun insertBackupConversationWithReadDate(
+        readDate: Instant,
+        conversationId: ConversationIDEntity? = null,
+        index: Int = Random.nextInt(0, 5)
+    ): ConversationEntity {
+        val randomID = Random.nextBytes(16).decodeToString()
+        val type = if (index % 2 == 0) ConversationEntity.Type.ONE_ON_ONE else ConversationEntity.Type.GROUP
+        val conversation = ConversationEntity(
+            id = conversationId ?: ConversationIDEntity(randomID, "some-domain-$index"),
+            name = "name-$index",
+            type = type,
+            teamId = null,
+            protocolInfo = ConversationEntity.ProtocolInfo.Proteus,
+            mutedStatus = ConversationEntity.MutedStatus.ALL_ALLOWED,
+            mutedTime = 0,
+            removedBy = null,
+            creatorId = "creatorId$index",
+            lastNotificationDate = UserDatabaseDataGenerator.DEFAULT_DATE,
+            lastModifiedDate = UserDatabaseDataGenerator.DEFAULT_DATE,
+            lastReadDate = readDate,
+            access = listOf(ConversationEntity.Access.values()[index % ConversationEntity.Access.values().size]),
+            accessRole = listOf(ConversationEntity.AccessRole.values()[index % ConversationEntity.AccessRole.values().size]),
+            receiptMode = ConversationEntity.ReceiptMode.DISABLED
+        )
+
+        backupDatabaseBuilder.conversationDAO.insertConversation(conversation)
+
+        return conversation
+    }
```

Comment:
UserDatabaseBuilder was meant to do such things like generting and inserting the test data so plx move it here otherwise we got a mess :S

Reply:
Updated